src/ir_passes/local_port_elim.rs

```diff
@@ -0,0 +1,63 @@
+use std::collections::HashMap;
+
+use crate::{
+    ir::{Access, Command, Component, Connect, Ctx, Info, PortIdx},
+    ir_visitor::{Action, Visitor},
+};
+
+/// Eliminates local ports from the IR.
+#[derive(Default)]
+pub struct LocalPortElim {
+    port_map: HashMap<PortIdx, PortIdx>,
+}
+
+impl Visitor for LocalPortElim {
+    /// loops over connect commands and adds local port connects to the mapping.
+    fn start(&mut self, comp: &mut Component) -> Action {
+        for (src, dst) in comp.cmds.iter().filter_map(|cmd| match cmd {
+            Command::Connect(Connect { src, dst, .. }) => {
+                Some((src.port, dst.port))
+            }
+            _ => None,
+        }) {
+            if comp.get(dst).is_local() {
+                self.port_map.insert(dst, src);
+            }
+        }
```

Comment:
These iterators can be replaced with `for_each` instead. Generally, mixing `for` and iterators makes the code pretty unreadable.